src/runtime_src/core/edge/common/aie_parser.cpp

```diff
@@ -158,6 +159,31 @@ get_gmio(const pt::ptree& aie_meta)
   return gmios;
 }
 
+std::vector<plio_type>
+get_plio(const pt::ptree& aie_meta)
+{
+  std::vector<plio_type> plios;
+
+  auto plio_nodes = aie_meta.get_child_optional("aie_metadata.PLIOs");
+  if (!plio_nodes)
+    return plios;
```

Comment:
You could do `return {}`
Then declare your vector before the for loop.

Reply:
done